### ISSUE TITLE
Input in data grid example is missing its accessible name

### ISSUE THREAD

**Issue by WilcoFiers** [w3c/aria-practices#2969]
On the data grid example (https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/#ex2_label), when you click the edit icon an text input element is provided to update the grid cell value. This is a text field has no accessible name. This seems required both by WCAG and by WAI-ARIA.

![Screenshot of axe extension highlighting the input field in the data grid example](https://github.com/w3c/aria-practices/assets/530687/f569da34-afed-46a6-94f5-40f7a3f7206c)
